### ISSUE TITLE
Question about KL Divergence loss function 关于KL散度的损失函数的问题

### ISSUE THREAD

**Issue by JunityZhan** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#469]
English and Chinese are translations to each other.

![2306be521dbeb70d2abf02686ca3ac38](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/assets/82944614/5f238455-5eaf-4d6d-8a94-c68b05ef3593)
This is loss function for KL Divergence. Specifically, for these formulas:
以下是KL散度的损失函数，特别是这段公式：
![579501e7e2a721be0c075efab22f2a4e](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/assets/82944614/92187cd7-f95e-4614-918f-caa88dd519fd)
they are acutally calculating kl divergence for gaussian distribution
他们其实是计算高斯分布的KL散度
![37b1f2ca39d43578c3ef97dcb429a064](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/assets/82944614/3a86875b-9222-476c-8a0a-00a44dc046c6)
But I find out it omits one term, which is $σ_1^2$
但是我发现他漏了一项$σ_1^2$
I don't know why it didn't write this term, if we include this term, the fomula should looked like this:
我不知道为什么他没写这一项，如果加上了这一项，这个公式应该长下面这样：
![08dde34439df32c28b3e25cc571340bb](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/assets/82944614/3aa0f935-b054-4223-b27e-f649260e6bdd)

Sincerey hoping for Answer.

**Comment by RVC-Boss** [comment]
You can also post this issue to 
https://github.com/jaywalnut310/vits

**Comment by JunityZhan** [comment]
> You can also post this issue to https://github.com/jaywalnut310/vits

Yeah, but I think there is no contributor maintaining that repo, so I want to try if some repos based on vits can answer it.

**Comment by RainaObi** [comment]
Very curious too. At first glance I thought about optimizations given the KL characteristics, but after scrambling over some papers about KL optimization and approximation I didn't find anything describing this case, or at least nothing that caught my attention.
Has anyone tried inserting the missing variance and comparing performance?

**Comment by JunityZhan** [comment]
> Very curious too. At first glance I thought about optimizations given the KL characteristics, but after scrambling over some papers about KL optimization and approximation I didn't find anything describing this case, or at least nothing that caught my attention. Has anyone tried inserting the missing variance and comparing performance?

I tried, and the performance is about the same. But the pretrained model must be trained on the version without that term. So I want to try training from the beginning, but I think it takes really long time.

**Comment by JunityZhan** [comment]
> Very curious too. At first glance I thought about optimizations given the KL characteristics, but after scrambling over some papers about KL optimization and approximation I didn't find anything describing this case, or at least nothing that caught my attention. Has anyone tried inserting the missing variance and comparing performance?

In addition, it calculate KL divergence by z_p, m_p, logs_q, logs_p, but z_p is the output of flow. In calculating KL divergence, it should be the mean of a distribution, but z_p definitely not a mean calculated by posterior encoder. (The mean of posterior encoder should be m_q, but not z_p)

**Comment by RainaObi** [comment]
Hi @JunityZhan, sorry for the ping. Did you learn something new regarding this issue from the other VITS based repos?

**Comment by JunityZhan** [comment]
> Hi @JunityZhan, sorry for the ping. Did you learn something new regarding this issue from the other VITS based repos?

Still no, I didn't find any related information and I think it is better to ask a professor but I can't because I am not a university student yet.😢

**Comment by sjkoelle** [comment]
I think the KL divergence may be with respect to a standard Gaussian so s_q is just 1?

**Comment by daniilrobnikov** [comment]
I have found this topic, where author discusses KL divergence in detail:
Please refer: [#6](https://github.com/jaywalnut310/vits/issues/6)

**Comment by p0p4k** [comment]
The KL divergence that author uses is correct. If you want to know more about it, please tag me or ping me on discord (p0p4k). I can explain it for anyone who is interested.
